### PR TITLE
Fix broken links

### PR DESCRIPTION
**Pull request by scheric**
**Description:**
Fixing some links.
 #10491

**Pull request in home-assistant (if applicable):** home-assistant/home-assistant#<home-assistant PR number goes here>

## Checklist:

- [ ] Branch: `next` is for changes and new documentation that will go public with the next Home Assistant release. Fixes, changes and adjustments for the current release should be created against `current`.
- [ ] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

**Comment by scheric** [comment]
@frenck  Should these PRs only have one component in it or should I create more PR.

**Comment by fabaff** [comment]
Usually smaller pull requests are preferred as they require less time to review.

File: source/_docs/scripts/conditions.markdown [hunk 1]
```diff
@@ -241,7 +241,7 @@ A better weekday condition could be by using the [Workday Binary Sensor](/integr
 
 ### Zone condition
 
-Zone conditions test if an entity is in a certain zone. For zone automation to work, you need to have setup a device tracker platform that supports reporting GPS coordinates. Currently this is limited to the [OwnTracks platform](/integrations/device_tracker.owntracks/) and the [iCloud platform](/integrations/device_tracker.icloud/).
+Zone conditions test if an entity is in a certain zone. For zone automation to work, you need to have setup a device tracker platform that supports reporting GPS coordinates. Currently this is limited to the [OwnTracks platform](/integrations/device_tracker.owntracks/) and the [iCloud platform](/integrations/icloud/).
```
Comment (fabaff) [inline source/_docs/scripts/conditions.markdown]: For stuff like that we have already [redirects](https://github.com/home-assistant/home-assistant.io/blob/next/source/_redirects#L339).

Comment (klaasnicolaas) [reply source/_docs/scripts/conditions.markdown]: Does not mean that this adjustment is not correct, once it has changed everywhere, the redirect is no longer necessary 😉

**Comment by klaasnicolaas** [comment]
What @fabaff already indicates, we prefer to have small PRs so that it is easy to review. But in this case it may be more than 1 link modification, if there are 15 then I think it's good too 😄

File: source/_docs/scripts/conditions.markdown (hunk 1 above)
Comment (scheric) [reply source/_docs/scripts/conditions.markdown]: But that means that the link should work.
If i click on [iCloud platform](https://www.home-assistant.io/integrations/device_tracker.icloud/) or [OwnTracks platform](https://www.home-assistant.io/integrations/device_tracker.owntracks/) in https://www.home-assistant.io/docs/scripts/conditions/#zone-condition I get redirected to Page Not Found.

Comment (scheric) [reply source/_docs/scripts/conditions.markdown]: Same with https://www.home-assistant.io/docs/mqtt/broker/#cloudmqtt
 [OwnTracks](https://www.home-assistant.io/integrations/device_tracker.owntracks)

Comment (frenck) [reply source/_docs/scripts/conditions.markdown]: While not broken, It is OK (and personally to me preferred) to change those. If we can reduce redirects, that is always a win. Both for client<->server round trips, but also SEO wise.

**Review by frenck — APPROVED** [review]
LGTM, Thanks @scheric!

Happy Hacktoberfest 🎉

**Pull request merged**